src/main/java/com/hivemq/adapter/sdk/api/config/UserProperty.java

```diff
@@ -15,39 +15,99 @@
  */
 package com.hivemq.adapter.sdk.api.config;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.hivemq.adapter.sdk.api.annotations.ModuleConfigField;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
 
 /**
  * Interface for parsing mqtt user properties in adapter configurations.
  * The implementation in the HiveMQ Edge code will be used to automatically create instances during conversion:
  * <p>
- *     \\@JsonProperty("propertyName")
- *     \\@ModuleConfigField(title = "Property Name", description = "Name of the associated property")
- *     private @Nullable String propertyName = null;
+ * \\@JsonProperty("propertyName")
+ * \\@ModuleConfigField(title = "Property Name", description = "Name of the associated property")
+ * private @Nullable String propertyName = null;
  * <p>
- *     \\@JsonProperty("propertyValue")
- *     \\@ModuleConfigField(title = "Property Value", description = "Value of the associated property")
- *     private @Nullable String propertyValue = null;
+ * \\@JsonProperty("propertyValue")
+ * \\@ModuleConfigField(title = "Property Value", description = "Value of the associated property")
+ * private @Nullable String propertyValue = null;
  */
-public interface UserProperty {
+public class UserProperty {
+
+    @JsonProperty("propertyName")
```

Comment:
nope. i looked it up and it was called the same before. I also wonder if this is good as it is a bit long. We could change it, but this would break old configs. Perhaps with an alias?

Reply:
I changed the json keys and added the alias annotation.